clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ArtifactController.java

```diff
@@ -66,4 +67,63 @@ StreamingResponseBody fetch(@RequestBody Artifact artifact) {
 
     return outputStream -> IOUtils.copy(artifactDownloader.download(artifact), outputStream);
   }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/{type}/account/{accountName}/names")
+  List<String> getNames(@PathVariable("type") String type,
+                        @PathVariable("accountName") String accountName) {
+    switch (type) {
+      case "helm":
+        return getHelmArtifactNames(accountName);
+      default:
+        throw new NotFoundException("Could not found artifact of type " + type);
+    }
+  }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/{type}/account/{accountName}/names/{artifactName}/versions")
+  List<String> getVersions(@PathVariable("type") String type,
+                           @PathVariable("accountName") String accountName,
+                           @PathVariable("artifactName") String artifactName) {
+    switch (type) {
+      case "helm":
+        return getHelmArtifactVersions(accountName, artifactName);
+      default:
+        throw new NotFoundException("Could not found artifact of type " + type);
+    }
+  }
+
+  private List<String> getHelmArtifactNames(String accountName) {
+    HelmArtifactCredentials helmArtifactCredentials = (HelmArtifactCredentials) findCredentials("helm/chart", accountName);
```

Comment:
You'll probably end up addressing this along with @lwander 's comment above (as it's related) but I'm not sure I like the necessity to cast from the base `ArtifactCredentials` to the specific credential type in the base controller, as well as the amount of helm-specific logic.

Ideally the specific downloading code here would be pushed into the `HelmArtifactCredentialsClass` and the controller would call a more general wrapper method rather than making specific calls like `downloadIndex` itself.

I actually wonder if we should even just expose `getArtifactNames` and `getArtifactVersions` in the `ArtifactCredentials` interface with default implementations that throw `Not implemented`, which will set us up for less code duplication if/when we do this for other types.